src/sync/index.js

```diff
@@ -95,40 +93,36 @@ async function sync(db) {
   } catch (err) {
     logger.error(`Error GET https://testnet.qtum.org/insight-api/status?q=getInfo: ${err.message}`);
   }
+
   sequentialLoop(
     Math.ceil((currentBlockChainHeight - startBlock) / batchSize), async (loop) => {
       const endBlock = Math.min((startBlock + batchSize) - 1, currentBlockChainHeight);
 
-      await syncTopicCreated(db, startBlock, endBlock, removeHexPrefix);
-      logger.debug('Synced Topics');
-
-      await Promise.all([
-        syncCentralizedOracleCreated(db, startBlock, endBlock, removeHexPrefix),
-        syncDecentralizedOracleCreated(db, startBlock, endBlock, removeHexPrefix, currentBlockTime),
-        syncOracleResultVoted(db, startBlock, endBlock, removeHexPrefix, oraclesNeedBalanceUpdate),
-      ]);
-      logger.debug('Synced Oracles');
-
-      await Promise.all([
-        syncOracleResultSet(db, startBlock, endBlock, removeHexPrefix, oraclesNeedBalanceUpdate),
-        syncFinalResultSet(db, startBlock, endBlock, removeHexPrefix, topicsNeedBalanceUpdate),
-      ]);
-      logger.debug('Synced Result Set');
-
-      const updateBlockPromises = [];
-      for (let i = startBlock; i <= endBlock; i++) {
-        const updateBlockPromise = new Promise(async (resolve) => {
-          await db.Blocks.insert({
-            _id: i,
-            blockNum: i,
-            blockTime: currentBlockTime,
-          });
-          resolve();
-        });
-        updateBlockPromises.push(updateBlockPromise);
+      try {
+        await syncTopicCreated(db, startBlock, endBlock, removeHexPrefix);
+        logger.debug('Synced Topics');
+
+        await Promise.all([
+          syncCentralizedOracleCreated(db, startBlock, endBlock, removeHexPrefix),
+          syncDecentralizedOracleCreated(db, startBlock, endBlock, removeHexPrefix, currentBlockTime),
+          syncOracleResultVoted(db, startBlock, endBlock, removeHexPrefix, oraclesNeedBalanceUpdate),
+        ]);
+        logger.debug('Synced Oracles');
+
+        await Promise.all([
+          syncOracleResultSet(db, startBlock, endBlock, removeHexPrefix, oraclesNeedBalanceUpdate),
+          syncFinalResultSet(db, startBlock, endBlock, removeHexPrefix, topicsNeedBalanceUpdate),
+        ]);
+        logger.debug('Synced Result Set');
+
+        const { insertBlockPromises, endBlockTime } = await getInsertBlockPromises(db, startBlock, endBlock);
+        await Promise.all(insertBlockPromises);
+        logger.debug('Inserted Blocks');
+
+        sendSyncInfo(endBlock, endBlockTime, chainBlockNum);
```

Comment:
this will cause a bunch of notification during init sync. I think during init sync, frontend should query progress periodically rather than relying on websocket. websocket is designed more for delta update

Reply:
Ya that was intentional. So you think we should just continuously poll for the syncInfo?

My intention was to just use a single `syncInfo` subscription for all syncInfo returns. If we continuously poll on an interval, doesn't it seems less efficient to use more network usage as we could poll the same syncInfo twice (or more) than being notified exactly when a block is done?